### DISCUSSION TITLE
Public domains and Deployment Protection

### DISCUSSION THREAD

**Discussion by Hildorien** [vercel/community#5108] [Help] [answered]
### Context:
I've encountered a challenge while working on a Vercel project linked to a GitHub repository with two branches: main and staging.
I'm new using Vercel and I have the Pro membership.

### Goal:
My goal is to establish two domains, each linked to a specific branch:

production-myawesomeapp.vercel.app -> linked to main
staging-myawesomeapp.vercel.app -> linked to staging
The staging domain should be public, allowing external engineers and APIs to access the application.

### Problem:
The project settings in Vercel permit linking only one branch to a production (public) domain. The resulting configuration is as follows:

production-webapp.vercel.app -> linked to main (Production)
staging-webapp.vercel.app -> linked to staging (Preview)

As staging-webapp.vercel.app is considered a preview domain, it's protected by the Vercel team. Consequently, only team members can access it or external individuals with a shareable link. This poses a challenge when testing external APIs that need to make requests to the staging domain since it's not public. 

I tried doing an Axios request from an external API hosted in AWS to the backend API of our NextJS app (hosted in the preview domain) and it fails because it needs authentication. With postman works fine because there is a cookie presetted (_vercel_jwt)

### Possible Solutions

1. Deployment Protection Features: 
Use the Deployment protection features, specifically Deployment protection exceptions or Protection bypass for automation, seems to be a potential solution. However, these features cost an additional $150.

2. Create a New Project: 
Another approach involves creating a new project, linking it to the same repository, and setting the production branch to be "staging." While this results in two separate production domains (public), it introduces the challenge of managing two projects that essentially refer to the same repository.

Has anyone encountered a similar challenge and found an alternative solution?

Are there any recommendations on effectively managing multiple domains for different branches within a single Vercel project?

### Steps to Reproduce

1. Create a project linked to a repository featuring two branches: main for production and staging/develop for development purposes.
2. Establish two domains within the same project to distinctly segregate production and development environments.
3. Connect the development branch to a public domain dedicated to sandbox testing.
4. Note that Vercel restricts the configuration of only one branch to a public domain, designating the other branch as a preview.
5. Recognize that the preview domain is inaccessible to external APIs by default, necessitating an investment of $150 for deployment protection features.
6. Evaluate whether to absorb the additional costs for enhanced functionality or opt for an alternative approach: 
Consider creating a new project linked to the same repository with the production branch set to "staging/develop," albeit managing two projects tied to a single repository.

**Comment by amyegan** [comment, marked as answer]
Hi @Hildorien. You can disable Vercel Authentication in project settings to allow public access to preview deployments.

<img width="887" alt="Project settings page showing " src="https://github.com/vercel/community/assets/8247105/91c674d3-4c3c-4bc0-91bf-2cd01abf52c7">

Please try that and let me know if you have any trouble with it 🙏

**Reply by Hildorien** [reply]
Yes! It worked. It was right in front of my eyes... Sorry!

**Reply by haiex509** [reply]
Thanks, it works.
